File: api/spec/factories/notice.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :notice do
    title { Faker::Book.title }
    text { Array.new(Random.rand(1..3)) { Faker::Books::Dune.quote }.join("\n") }
    pinned { Random.rand(4).zero? }
    team { nil }
    created_at { Time.current - Random.rand(60).minutes - Random.rand(60).seconds }
    updated_at { created_at }
  end
end

File: ui/orm/Notice.js
import orm from '~/orm'
import BaseModel from '~/orm/BaseModel'

export default class Notice extends BaseModel {
  static entity = 'notices'

  static fields() {
    return {
      id: this.string(),
      title: this.string(),
      text: this.string(),
      pinned: this.boolean(),
      teamId: this.string().nullable(),
      team: this.belongsTo(orm.Team, 'teamId'),
      createdAt: this.string()
    }
  }
}
